otter/convergence/planning.py

```diff
@@ -367,6 +373,103 @@ def drain_and_delete_a_server(server):
                 converge_later)
 
 
+def converge_launch_stack(desired_state, stacks):
+    """
+    Create steps that indicate how to transition from the state provided
+    by the given parameters to the :obj:`DesiredStackGroupState` described by
+    ``desired_state``.
+
+    :param DesiredStackGroupState desired_state: The desired group state.
+    :param set stacks: a set of :obj:`HeatStack` instances.
+        This must only contain stacks that are being managed for the specified
+        group.
+    :rtype: :obj:`pbag` of `IStep`
+
+    """
+    config = desired_state.stack_config
+
+    group_states = [StackState.CREATE_UPDATE_COMPLETE,
```

Comment:
I think this code could be a bit more clear by just doing something like

```python
stacks_by_state = groupby(lambda stack: stack.get_state(), stacks)
stacks_complete = stacks_by_state.get(StackState.CREATE_UPDATE_COMPLETE, [])
stacks_failed = stacks_by_state.get(StackState.CREATE_UPDATE_FAILED, [])
```

This would be significantly shorter (line-wise), and I think more obvious to casual readers.

Anyway, just a nitpick, feel free to ignore :)